rdpy/core/type.py

```diff
@@ -916,7 +925,26 @@ def writeType(self, value):
                 self.writeType(element)
             return
         value.write(self)
-        
+
+
+class ByteStream(BytesIO):
+    """
+    @summary:  ByteStream used to write all types, inherits BytesIO
+    """
+
+    def writeType(self, value):
+        """
+        @summary:  Call specific write on type object
+                    or iterate over tuple element
+        @param value: (tuple | Type)
+        """
+        if isinstance(value, tuple) or isinstance(value, list):
```

Comment:
`isinstance(value, (tuple, list))` would be more compact. It might be better to use the abstract base class `Sequence` and explicitly filter out strings:
```python
from collections.abc import Sequence  # Off the top of my head
isinstance(value, Sequence) and not isinstance(value, basestring)  # Just `str` in Python 3
```